server/src/main/resources/db/changelog/20170718095058-purge-orphaned-pools.xml

```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+
+    <changeSet id="20170718095058-1" author="vrjain">
+        <comment> purge orphaned pools </comment>
+        <sql>
+            delete from cp_pool where id in
+            (  select p.id
+               from cp_pool p
+                   left outer join cp_pool_source_stack ss
+               on p.id = ss.derivedpool_id
+               where p.type='STACK_DERIVED'
+                 and ss.derivedpool_id is null
```

Comment:
This inner query will never match anything. We're joining on ss.derivedpool_id (which will never match using equals if the value is null), but then checking that the value is null.